include/bio_ik/goal_types.h

```diff
@@ -297,6 +297,38 @@ class LineGoal : public LinkGoalBase
     }
 };
 
+class PlaneGoal : public LinkGoalBase
+{
+    tf2::Vector3 position;
+    tf2::Vector3 normal;
+
+public:
+    PlaneGoal()
+        : position(0, 0, 0)
+        , normal(0, 0, 1)
+    {
+    }
+    PlaneGoal(const std::string& link_name, const tf2::Vector3& position, const tf2::Vector3& normal, double weight = 1.0)
+        : LinkGoalBase(link_name, weight)
+        , position(position)
+        , normal(normal.normalized())
+    {
+    }
+    const tf2::Vector3& getPosition() const { return position; }
+    void setPosition(const tf2::Vector3& p) { position = p; }
+    const tf2::Vector3& getNormal() const { return normal; }
+    void setNormal(const tf2::Vector3& d) { normal = d.normalized(); }
+    virtual double evaluate(const GoalContext& context) const
+    {
+        // Get the position of the link
+        auto link_position = context.getLinkFrame().getPosition();
+        // Get the signed distance from the link to the plane
+        double signed_dist = (link_position - position).dot(normal);
+        // Take the squared value of the signed distance
+        return std::pow(signed_dist, 2);
```

Comment:
Please avoid `std::pow`. Yes, compilers will usually optimize it away, [unless you explicitly disable optimizations](https://godbolt.org/z/cKxjoK53E), but the rest of the code-base does not use it either.
Also, no need for the copilot-style comments that squaring squares and that `link_position` is the position of the link. :)